packages/promise-kit/index.js

```diff
@@ -45,6 +45,9 @@ harden(makePromiseKit);
  * @returns {Promise<Awaited<T>>} A new Promise.
  */
 export function racePromises(values) {
-  return harden(memoRace.call(BestPipelinablePromise, values));
+  const p = /** @type {Promise<Awaited<T>>} */ (
+    memoRace.call(BestPipelinablePromise, values)
+  );
+  return harden(p);
```

Comment:
One line of nice JS turns into 4 ugly lines just to please TypeScript, why isn't it possible that TS properly infers typs from `.call` :disappointed: 

I'm all for strict types, but TypeScript just doesn't support it itself... so it feels kinda pointless to ts-ignore everything?